R/helper_functions.R

```diff
@@ -82,18 +83,19 @@ prepareComboMergedData <- function(cell_lines,
 
 #' prepareCodilutionData
 #' 
-#' Create data.table with input codilution data containg noise for testing purpses 
+#' Create data.table with input co-dilution data containing noise for testing purposes 
 #'
 #' @param df data.table object with experiment data
-#' @param df_layout data.table object with experiment design containg drugs and cell line data
+#' @param df_layout data.table object with experiment design containing drugs and cell line data
 #' @return data.table with input data for testing
 #' 
 #' @examples
 #' 
-#' df_layout <- prepareData(create_synthetic_cell_lines()[seq_len(2), ],
-#' create_synthetic_drugs()[seq_len(4), ])
-#' df <- cbind(create_synthetic_cell_lines()[1, , drop = FALSE],
-#' df_layout[, "Concentration", drop = FALSE])
+#' cell_lines <- create_synthetic_cell_lines()
+#' drugs <- create_synthetic_drugs()
+#' df_layout <- prepareData(cell_lines[seq_len(2), ], drugs[seq_len(4), ])
+#' df <- cbind(drugs[1, , drop = FALSE],
+#'             df_layout[, "Concentration", drop = FALSE])
```

Comment:
@gladkia @bczech 
I was wondering, as user I would expect the same skeleton for all function creating synthetic full respons data. 
So, as `prepareData`, `prepareMergedData` and `prepareComboMergedData` as basic params are using 
```
cell_lines <- create_synthetic_cell_lines()
drugs <- create_synthetic_drugs()
```
with some additional specyfic params (`noise`, `concentration`, etc.)

*Also `prepareCodilutionData` could be build in the same way.*

My proposal:
```
#' prepareCodilutionData
#' 
#' Create data.table with input co-dilution data containing noise for testing purposes 
#'
#' @param cell_lines data.table with cell line info
#' @param drugs data.table with drug info
#' @param conc vector of doses
#' 
#' @return data.table with input data for testing
#' 
#' @examples
#' 
#' prepareCodilutionData(create_synthetic_cell_lines()[seq_len(2), ],
#'                       create_synthetic_drugs()[seq_len(4), ])
#' 
#' @export
prepareCodilutionData <- function(cell_lines, drugs, conc = 10 ^ (seq(-3, 1, 0.5)))  {

  df_layout <- prepareData(cell_lines = cell_lines, drugs = drugs, conc = conc)
  df <- cbind(drugs[1, , drop = FALSE],
              df_layout[, "Concentration", drop = FALSE])
  
  colnames(df) <- paste0(colnames(df), "_2")
  df_2 <- cbind(df_layout, df)
  df_2 <- df_2[df_2$DrugName != df_2$DrugName_2, ]
  rows <- which(df_2$Concentration_2 > 0)
  cols <- c("Concentration", "Concentration_2")
  df_2[rows, (cols) := lapply(.SD, function(x) x / 2), .SDcols = cols]
  df_2
}
```

```

Reply:
Agree, but I wouldn't subset the first row of the drugs data.table  in the function body

Reply:
Why?

This subset of the firs row creates second drug
```
  df <- cbind(drugs[1, , drop = FALSE],
              df_layout[, "Concentration", drop = FALSE])

> df |> head()
   Gnumber_2 DrugName_2 drug_moa_2 Concentration_2
1:    G00001   drug_001      moa_A               0
2:    G00001   drug_001      moa_A               0
3:    G00001   drug_001      moa_A               0
4:    G00001   drug_001      moa_A               0
5:    G00001   drug_001      moa_A               0
6:    G00001   drug_001      moa_A               0
```

@bczech What you proposed instead?

Reply:
@bczech @gladkia 
In fact that should be like that - so to achieve consistent form also `noise` and part with `generate_response_data()` have to be added

```
#' prepareCodilutionData
#' 
#' Create data.table with input co-dilution data containing noise for testing purposes 
#'
#' @param cell_lines data.table with cell line info
#' @param drugs data.table with drug info
#' @param conc vector of doses
#' @param noise number indicating level of noise
#' 
#' @return data.table with input data for testing
#' 
#' @examples
#' 
#' prepareCodilutionData(create_synthetic_cell_lines()[seq_len(2), ],
#'                       create_synthetic_drugs()[seq_len(4), ])
#' 
#' @export
prepareCodilutionData <- function(cell_lines, 
                                  drugs, 
                                  conc = 10 ^ (seq(-3, 1, 0.5)),
                                  noise = 0.1) {

  df_layout <- prepareData(cell_lines = cell_lines, drugs = drugs, conc = conc)
  
  df_2 <- cbind(drugs[1, , drop = FALSE],
              df_layout[, "Concentration", drop = FALSE])
  colnames(df_2) <- paste0(colnames(df_2), "_2")
  
  df_layout_2 <- cbind(df_layout, df_2)
  df_layout_2 <- df_layout_2[df_layout_2$DrugName != df_layout_2$DrugName_2, ]
  rows <- which(df_layout_2$Concentration_2 > 0)
  cols <- c("Concentration", "Concentration_2")
  df_layout_2[rows, (cols) := lapply(.SD, function(x) x / 2), .SDcols = cols]

  generate_response_data(df_layout_2, noise)
}
```

Reply:
It looks good to me.

Reply:
Above change influence only one other function
https://github.com/gdrplatform/gDRcore/blob/master/R/generate_wrappers.R#L317-L338 
```
df_merged <- prepareCodilutionData(cell_lines[seq(1, 15, 2), ], drugs[seq_len(12), ])
```

Reply:
I propose not to subsetting first row from drugs, but to use all rows. In the presented scenario we assume that the user always will want to use the first drug as a second drug to build codilution. Or even safe would be to add function argument that would take a numeric vector for submitting drug table – since in gDRcore we use it either:
https://github.com/gdrplatform/gDRcore/blob/33cdbc624a8739716ce230c8e5338c486692c724/R/generate_wrappers.R#L294
or 
https://github.com/gdrplatform/gDRcore/blob/33cdbc624a8739716ce230c8e5338c486692c724/R/generate_wrappers.R#L321

Reply:
1. the firs row was proposed as a subset in all location where this function was used
2. with proposed param
```
#' prepareCodilutionData
#' 
#' Create data.table with input co-dilution data containing noise for testing purposes 
#'
#' @param cell_lines data.table with cell line info
#' @param drugs data.table with drug info
#' @param drugsIdx2 numeric index of ids for secondary drug
#' @param conc vector of doses
#' @param noise number indicating level of noise
#' 
#' @return data.table with input data for testing
#' 
#' @examples
#' 
#' prepareCodilutionData(create_synthetic_cell_lines()[seq_len(2), ],
#'                       create_synthetic_drugs()[seq_len(4), ])
#' 
#' @export
prepareCodilutionData <- function(cell_lines, 
                                  drugs, 
                                  drugsIdx2 = 1,
                                  conc = 10 ^ (seq(-3, 1, 0.5)),
                                  noise = 0.1) {

  df_layout <- prepareData(cell_lines = cell_lines, drugs = drugs, conc = conc)
  
  df_2 <- cbind(drugs[drugsIdx2, , drop = FALSE],
              df_layout[, "Concentration", drop = FALSE])
  colnames(df_2) <- paste0(colnames(df_2), "_2")
  
  df_layout_2 <- cbind(df_layout, df_2)
  df_layout_2 <- df_layout_2[df_layout_2$DrugName != df_layout_2$DrugName_2, ]
  rows <- which(df_layout_2$Concentration_2 > 0)
  cols <- c("Concentration", "Concentration_2")
  df_layout_2[rows, (cols) := lapply(.SD, function(x) x / 2), .SDcols = cols]

  generate_response_data(df_layout_2, noise)
}
```

Reply:
I prefer the latter solution.

Reply:
It will be done in separate task.